lib/game_team.rb

```diff
@@ -144,6 +144,47 @@ def self.gets_team_shots_goals_count(season)
     stats_by_team = get_goal_shots_by_game_team(matches)
   end
 
+#Michelle Start
+  def self.games_ids_by_season(season_id)
+    game_id_first = season_id.to_s[0..3].to_i
+    all_games_by_id = all.find_all { |game| game.season_id == game_id_first }
+    all_games_by_id.map { |game| game.game_id }
+  end
+
+  def self.games_by_season_id(season_id)
+      games_by_id = []
+      all.each do |game|
+        if games_ids_by_season(season_id).any? { |id| id == game.game_id }
+          games_by_id << game
+        end
+      end
+      games_by_id
+  end
+
+  def self.games_by_team_name(season_id)
```

Comment:
This method could probably be incorporated into tackles_by_team method since thats the only one that uses it!

Reply:
agreed, good catch